lib/modules/manager/vendir/artifacts.ts

```diff
@@ -0,0 +1,126 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
+import { exec } from '../../../util/exec';
+import type { ExecOptions, ToolConstraint } from '../../../util/exec/types';
+import {
+  getParentDir,
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+import { isFileInDir } from './utils';
+
+async function vendirCommands(execOptions: ExecOptions): Promise<void> {
+  const cmd: string[] = [];
+
+  cmd.push(`vendir sync`);
+
+  await exec(cmd, execOptions);
+}
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`vendir.updateArtifacts(${packageFileName})`);
+
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+  const isUpdateOptionAddVendirArchives = config.postUpdateOptions?.includes(
+    'vendirUpdateSubChartArchives'
+  );
+
+  if (
+    !isLockFileMaintenance &&
+    (updatedDeps === undefined || updatedDeps.length < 1)
+  ) {
+    logger.debug('No updated vendir deps - returning null');
+    return null;
+  }
+
+  const lockFileName = getSiblingFileName(packageFileName, 'vendir.lock.yml');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent && !isUpdateOptionAddVendirArchives) {
+    logger.debug('No vendir.lock.yml found');
+    return null;
+  }
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
+    logger.debug('Updating Vendir artifacts');
+    const execOptions: ExecOptions = {
+      cwdFile: packageFileName,
+      toolConstraints: [
+        { toolName: 'vendir', constraint: config.constraints?.vendir },
```

Comment:
is this supported on containerbase/base image?

Reply:
Yes. I've confirmed that